src/harness/BenchmarkDotNet.Extensions/ValuesGenerator.cs

```diff
@@ -128,6 +128,8 @@ public static string[] ArrayOfStrings(int count, int minLength, int maxLength)
 
         private static T GenerateValue<T>(Random random)
         {
+            if (typeof(T) == typeof(byte))
+                return (T)(object)(byte)random.Next(byte.MinValue, byte.MaxValue);
```

Comment:
Technically this is never going to generate a return value of `byte.MaxValue` because `Random.Next` upper limit is exclusive, but we're not obsessing about full coverage in consumers of  this helper.

Reply:
Why not just add one to the upper bound in those cases?

Reply:
Certainly possible that would work, just not consistent with the existing behavior... I'll respect any decision on that, just let me know.

Reply:
I bet the existing cases (short, char etc) are not missing a +1 intentionally. As you say, it may not matter, but it would be the expectation of a naive caller. @adamsitnik  can say.

Reply:
> Random.Next upper limit is exclusive

To be honest we just missed it when this was being implemented. You can change it for the new types that are being added here, but we can't change it for types that are already supported. But most likely keeping the consistency is going to be the best.

Why we can't change it for the types that were already supported? https://github.com/dotnet/performance/blob/main/docs/microbenchmark-design-guidelines.md#always-the-same-input-data So if we change it, a method that was generating `int[]` for `Array.Sort` could start generating different data (example: `2 int.MaxValue 1` instead `1 3 2` ) and result in the benchmarks doing more/less work and showing up as a performance regression/improvement.

Reply:
Added comments to the code to warn NOT changing things that have been public before.